tests/test_utils.py

```diff
@@ -606,19 +606,22 @@ def test_etag_strong_vs_weak_comparison(self):
         assert not weak_67aB43.strong_compare(weak_67ab43_one)
 
     @pytest.mark.parametrize(
-        'filename,expected',
+        'filename,max_length,expected',
         [
-            ('.', '_'),
-            ('..', '_.'),
-            ('hello.txt', 'hello.txt'),
-            ('Ąžuolai žaliuos.jpeg', 'A_z_uolai_z_aliuos.jpeg'),
-            ('/etc/shadow', '_etc_shadow'),
-            ('. ⬅ a dot', '____a_dot'),
-            ('C:\\Windows\\kernel32.dll', 'C__Windows_kernel32.dll'),
+            ('.', None, '_'),
+            ('..', None, '_.'),
+            ('hello.txt', None, 'hello.txt'),
+            ('hellotxt', 6, 'hellotxt'),
+            ('Ąžuolai žaliuos.jpeg', None, 'A_z_uolai_z_aliuos.jpeg'),
+            ('/etc/shadow', None, '_etc_shadow'),
+            ('. ⬅ a dot', None, '____a_dot'),
+            ('C:\\Windows\\kernel32.dll', None, 'C__Windows_kernel32.dll'),
+            ('Ąžuolai žaliuos.jpeg', 10, 'A_z_u.jpeg'),
```

Comment:
let's add tests where the ext length is
- greater than the limit
- equal to the limit

Reply:
Added more tests of different kind